### PR TITLE
COCOS-73 - Remove server

### LINKED ISSUE #73: Manager should be client to external gRPC server
Manager will establish a connection when started
Bi-directional streaming of agent events, agent logs after sending who am I request and receiving a run request (computation manifest)

### PR DESCRIPTION
**Pull request by SammyOina**
The manager service and client have been restructured for stream communication, facilitating real-time agent events, logs, and run responses. The `Run` RPC is replaced by the `Process` stream RPC, enabling bidirectional streaming between clients and the manager service. This allows continuous interchange of different message types including `WhoAmIRequest`, `AgentLog`, `AgentEvent`, and `RunResponse`.

Several message types have been adjusted and new fields introduced, like `AgentPort` in `RunResponse` and various agent-config attributes including CA files and instance IDs, to support TLS client authentication and distinguish between agent instances.

We've also incorporated `google.protobuf.Timestamp` in `AgentEvent` for precise event logging. The client code reflects these modifications with updated method calls and stream handling logic for ongoing communication. Moreover, the updates necessitate corresponding changes throughout service, grpc, and sdk layers to interoperate with the new streaming approach.

The transition to streaming paves the way for a more interactive, flexible communication system that can accommodate future expansion and real-time monitoring features.

Resolves #73

### CODE REVIEW

File: manager/service.go [hunk 1]
```diff
@@ -62,20 +62,19 @@ func New(qemuCfg qemu.Config, logger *slog.Logger, eventSvc events.Service, host
 	return ms
 }
 
-func (ms *managerService) Run(ctx context.Context, c *Computation) (string, error) {
+func (ms *managerService) Run(ctx context.Context, c *ComputationRunReq) (string, error) {
```
Comment (WashingtonKK) [inline manager/service.go:65]: Remove context

Comment (SammyOina) [reply manager/service.go:65]: required for tracing

**Review by WashingtonKK — APPROVED** [review]
LGTM

File: cmd/manager/main.go [hunk 2]
```diff
@@ -107,27 +103,35 @@ func main() {
...
-		log.Printf("failed to load %s gRPC server configuration: %s", svcName, err.Error())
+	pc, err := managerClient.Process(ctx)
+	if err != nil {
+		logger.Error(err.Error())
 		return
 	}
-	registerManagerServiceServer := func(srv *grpc.Server) {
-		reflection.Register(srv)
-		manager.RegisterManagerServiceServer(srv, managergrpc.NewServer(svc))
+	if err := pc.Send(&manager.ClientStreamMessage{Message: &manager.ClientStreamMessage_WhoamiRequest{}}); err != nil {
```
Comment (drasko) [inline cmd/manager/main.go:124]: Is this Whoami request really requests something, or is is just the notifications? Maybe I would just call it Whoami(), without `Request`.

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**